src/display/annotation_layer.js

```diff
@@ -718,6 +724,78 @@ var StrikeOutAnnotationElement = (
 })();
 
 /**
+ * @class
+ * @alias FileAttachmentAnnotationElement
+ */
+var FileAttachmentAnnotationElement = (
+    function FileAttachmentAnnotationElementClosure() {
+  function FileAttachmentAnnotationElement(parameters) {
+    var isRenderable = !!(parameters.data.hasPopup ||
+                          parameters.data.title || parameters.data.contents);
+    AnnotationElement.call(this, parameters, isRenderable);
+
+    this.filename = getFilenameFromUrl(parameters.data.file.filename);
+    this.content = parameters.data.file.content;
+  }
+
+  Util.inherit(FileAttachmentAnnotationElement, AnnotationElement, {
+    /**
+     * Render the file attachment annotation's HTML element in the empty
+     * container.
+     *
+     * @public
+     * @memberof FileAttachmentAnnotationElement
+     * @returns {HTMLSectionElement}
+     */
+    render: function FileAttachmentAnnotationElement_render() {
+      this.container.className = 'fileAttachmentAnnotation';
+
+      var trigger = document.createElement('div');
+      trigger.style.height = this.container.style.height;
+      trigger.style.width = this.container.style.width;
+      trigger.addEventListener('click', this._download.bind(this));
```

Comment:
Having played a bit with the code, I'm not sure why `dblclick` doesn't work here!?

Perhaps I'm missing something, but currently after clicking on the trigger and downloading the attachment, I need to click on the *popup* to close it thus introducing one more step into the work-flow (and if the user accidentally clicks on the trigger instead, the download is offered again).

If `dblclick` is used instead, then the popup is "automatically" closed when the attachment is downloaded, since the double-click will first show and the hide the popup.